timelock-impl/src/test/java/com/palantir/atlasdb/timelock/batch/MultiClientConjureTimelockResourceTest.java

```diff
@@ -67,59 +58,23 @@ public void before() {
         assertThat(Futures.getUnchecked(resource.leaderTimes(AUTH_HEADER, namespaces)))
                 .isEqualTo(getLeaderTimesForNamespaces(namespaces));
     }
 
-    @Test
-    public void canGetCommitTimestampsForMultipleClients() {
-        GetCommitTimestampsResponse getCommitTimestampsResponse = GetCommitTimestampsResponse.of(
-                COMMIT_TS_LOWER_INCLUSIVE, COMMIT_TS_UPPER_INCLUSIVE, lockWatchStateUpdate);
-
-        when(timelockService.getCommitTimestamps(anyInt(), any()))
-                .thenReturn(Futures.immediateFuture(getCommitTimestampsResponse));
-
-        Set<String> namespaces = ImmutableSet.of("client1", "client2");
-        assertThat(Futures.getUnchecked(
-                        resource.getCommitTimestamps(AUTH_HEADER, getGetCommitTimestampsRequests(namespaces))))
-                .isEqualTo(getGetCommitTimestampsResponseList(namespaces));
-    }
-
     @Test
     public void requestThrowsIfAnyQueryFails() {
         when(timelockService.leaderTime())
                 .thenReturn(Futures.immediateFuture(leaderTime))
                 .thenThrow(new BlockingTimeoutException(""));
-        Set<String> namespaces = ImmutableSet.of("client1", "client2");
+        Set<Namespace> namespaces = ImmutableSet.of(Namespace.of("client1"), Namespace.of("client2"));
         assertThatThrownBy(() -> Futures.getUnchecked(resource.leaderTimes(AUTH_HEADER, namespaces)))
                 .isInstanceOf(BlockingTimeoutException.class);
     }
 
-    private List<NamespacedGetCommitTimestampsResponse> getGetCommitTimestampsResponseList(Set<String> namespaces) {
-        return namespaces.stream()
-                .map(namespace -> NamespacedGetCommitTimestampsResponse.builder()
-                        .namespace(namespace)
-                        .inclusiveLower(COMMIT_TS_LOWER_INCLUSIVE)
-                        .inclusiveUpper(COMMIT_TS_UPPER_INCLUSIVE)
-                        .lockWatchUpdate(lockWatchStateUpdate)
-                        .build())
-                .collect(Collectors.toList());
-    }
-
-    private List<NamespacedGetCommitTimestampsRequest> getGetCommitTimestampsRequests(Set<String> namespaces) {
-        return namespaces.stream()
-                .map(namespace -> NamespacedGetCommitTimestampsRequest.builder()
-                        .namespace(namespace)
-                        .numTimestamps(4)
-                        .build())
-                .collect(Collectors.toList());
-    }
-
-    private List<NamespacedLeaderTime> getLeaderTimesForNamespaces(Set<String> namespaces) {
-        return namespaces.stream()
-                .map(namespace -> NamespacedLeaderTime.of(namespace, leaderTime))
-                .collect(Collectors.toList());
+    private LeaderTimes getLeaderTimesForNamespaces(Set<Namespace> namespaces) {
+        return LeaderTimes.of(namespaces.stream().collect(Collectors.toMap(x -> x, x -> leaderTime)));
```

Comment:
Hmm. I think this test really should target _two_ underlying timelock services which return _different_ times, and we should check that the wiring is correct. (This test would allow for a couple of bad behaviours!)